### ISSUE TITLE
“Get Status” iOS shortcut action does not return exit node

### ISSUE THREAD

**Issue by aegjoyce** [tailscale/tailscale#10735]
### What is the issue?

Using the “Get Status” node in iOS shortcuts to get the string for “Exit Node Name” does not work correctly and always returns the string “Tailscale” even when using a named exit node.

### Steps to reproduce

Connect to an exit node. Create a shortcut in iOS and add a “Get Status” node. Add a “Quick Look” node and change its input from “Status” to “Exit Node Name”. Run the shortcut - it will always return “Tailscale” as the string. 

### Are there any recent changes that introduced the issue?

N/A

### OS

iOS

### OS version

iOS 17.2

### Tailscale version

1.56.1

### Bug report

BUG-42ab706927c8506fcafff376af75d90dd2026a5eb8e533f22f978c7cfadfc407-20240103212627Z-6f5041ee3f1dd95c

**Comment by agottardo** [comment]
Cannot repro on 21D5026f and Tailscale 1.57.44.

Is this a shortcut you created on an older version of Tailscale (v1.50 or earlier)?

<img src="https://github.com/tailscale/tailscale/assets/9057073/e082a81e-9444-4ae4-99f7-98378c8454ef" width="250" />

**Comment by aegjoyce** [comment]
Nope, created on 1.50.1 - I did just find that my exit nodes had not updated in a while but having updated them all to 1.56.1 I'm still getting the same issue.

On further investigation - it looks like the shortcut is ignoring my custom machine names and instead using the auto-generated machine names instead. I have two exit nodes which on all my instances of Tailscale appear with the names I have assigned them in the dashboard. However the shortcut is displaying the auto-generated names (taken from the hostname) instead ("Tailscale" and "raspberrypi" in my case)

**Comment by agottardo** [comment]
If you create an entirely new shortcut from scratch, does the issue persist? In 1.50 we switched from the legacy SiriKit Intents to the new App Intents framework. I'm trying to understand if this is an issue with the migration process in the Shortcuts app.

**Comment by aegjoyce** [comment]
It does still persist, yes

**Comment by agottardo** [comment]
@aegjoyce can you send me a sysdiagnose file taken from your iPhone? You can email it to `andrea (at) tailscale (dot) com`.

**Comment by aegjoyce** [comment]
I can’t unfortunately (complex reasons why) but happy to help test in other ways if feasible?

**Comment by agottardo** [comment]
I think there is something about your shortcut or a bug in the Shortcuts editor that is causing the wrong field of our output content item to be passed to the Show in Quick Look action. 

Nowhere in our code we return "Tailscale" as an exit node name. On our side, we either return a nil value, or the exit node name. No other value is possible. "Tailscale" is the name of our app, and I suspect something is causing Shortcuts to create a string representation of the content item by using our app name. So this is most likely an issue with the Shortcuts app, and that is why I asked for a sysdiagnose, I wanted to see if the Shortcuts logs can help us track down the issue. 

I recommend filing a feedback with Apple if you're unable to provide diagnostics here (and unfortunately for you, they'll also ask you to provide a sysdiagnose to investigate this).

**Comment by aegjoyce** [comment]
Sorry, I should have been more clear - the string 'tailscale' is being returned as my exit node name because it's the hostname of my proxmox lxc that I'm running Tailscale on. Similarly, on another exit node, 'raspberrypi' is being returned as the exit node name (again, the hostname of that device).

The issue is that the shortcut is returning the default name (the hostname) of the exit node, rather than the custom name I assign for each exit node in the tailscale admin dashboard

Hope this makes more sense!

**Comment by agottardo** [comment]
Ah, that definitely changes things. I will update this issue accordingly.
